pkg/vcs/vcs.go

```diff
@@ -58,3 +87,86 @@ func VCS(path string) (result v1.VCS) {
 
 	return
 }
+
+func ImageInfoFromApp(ctx context.Context, app *apiv1.App, cloneDir string) (string, string, error) {
+	vcs := app.Status.AppImage.VCS
+	if len(vcs.Remotes) == 0 {
+		return "", "", fmt.Errorf("clone can only be done on an app built from a git repository")
+	}
+	if vcs.Acornfile == "" || vcs.BuildContext == "" {
+		return "", "", fmt.Errorf("app is missing required vcs information, image must be rebuilt with a newer acorn cli")
+	}
+
+	// Create auth object to use when fetching and cloning git repos
+	auth, err := ssh.NewSSHAgentAuth("git")
+	if err != nil {
+		return "", "", err
+	}
+
+	for _, remote := range vcs.Remotes {
+		// Since we use ssh auth to clone the repo we need a git url but will sometimes get http urls
+		var gitUrl string
+		httpUrl, err := url.Parse(remote)
+		if err == nil {
+			gitUrl = fmt.Sprintf("git@%s:%s", httpUrl.Host, httpUrl.Path[1:])
+		} else {
+			gitUrl = remote
+		}
+
+		// Determine the repository name from the repo url
+		idx := strings.LastIndex(remote, "/")
+		if idx < 0 || idx >= len(remote)-1 {
+			fmt.Printf("failed to determine repository name %q\n", remote)
+			continue
+		}
+		workdir := filepath.Join(cloneDir, strings.TrimSuffix(remote[idx+1:], ".git"))
+
+		// Clone git repo and checkout revision
+		fmt.Printf("# Cloning repository %q into directory %q\n", gitUrl, workdir)
+		repo, err := git.PlainCloneContext(ctx, workdir, false, &git.CloneOptions{
+			URL:      gitUrl,
+			Auth:     auth,
+			Progress: os.Stderr,
+		})
+		if err != nil {
+			fmt.Printf("failed to clone repository %q: %s\n", gitUrl, err.Error())
```

Comment:
Should we check for a `git.ErrRepositoryAlreadyExists` error here and bail if you get it? No other clone attempt will work, I think.

Reply:
I pulled this branch and tested a few things myself. Seems to work well. This is the only comment I have at this point.

Reply:
Also, in general, I think it is cleaner to do:
```
fmt.Printf("failed to clone repository %q: %v\n", gitUrl, err)
```